paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java

```diff
@@ -155,7 +152,17 @@ private void retractWithSequenceGroup(KeyValue kv) {
                             row.setField(i, getters[i].getFieldOrNull(kv.value()));
                         } else {
                             // retract normal field
-                            row.setField(i, null);
+                            if (kv.valueKind() == RowKind.UPDATE_BEFORE) {
```

Comment:
We should retract for delete too.

Reply:
I think only agg#retract is not enough, I think we should have 3 strategy for the retract record (DELETE/UPDATE_BEFORE).

And each aggregate function will be assigned with a retract strategy.

- `ignore_retract`  if user enabled `fields.${filed}.ignore-retract`
- `set_null`  for the last_value / last_not_null_value  user can use this to handle the retracted value to set null. This should be default for the field which is not declared as an agg field in partial-update for compatibility
- default call `agg#retract`

What do you think @JingsongLi

Reply:
Yes, correct